### DISCUSSION TITLE
can I base validation rules on the selection of radio buttons?

### DISCUSSION THREAD

**Discussion by julisch94** [vueform/vueform#127] [Questions] [answered]
Hey folks! Thanks a lot for this great library!

I have a contact form that contains a radio group (email, phone, messenger) and then 2 text fields, one for the email address and one for the phone number.

What I want:
- when the user selects 'phone', I want the phone number field to be required
- when the user selects 'email', I want the email address field to be required
- when the user selects both, I want both fields to be required

<img width="856" alt="image" src="https://github.com/vueform/vueform/assets/22644876/ebd13008-d082-4919-b986-19cbe0cc4281">

I was looking at the API for `rules` (https://vueform.com/reference/generic-element#option-rules) but it doesn't seem to handle more than strings.

Is there a way to achieve what I'm aiming for?

**Comment by adamberecz** [comment, marked as answer]
I think what you're looking for are [conditional rules](https://vueform.com/docs/validating-elements#conditional-rules).

**Reply by julisch94** [reply]
Oh great! Thanks! I was able to do this and it worked 🚀 
```html
<TextElement name="phone" input-type="tel" label="Telefonnummer" :rules="[
    {
      required: ['checkboxgroup', ['phone', 'whatsapp']]
    }
  ]" />
```

**Reply by julisch94** [reply]
sorry @adamberecz to bring this back up, but how can I combine the `required` from above with a `phone` or `email` rule?

Example: 
- when the user has selected phone, I want the phone field to be required AND be a valid phone number
- when the user has selected email, I want the email field to be required AND be a valid email address

I've found the [AND Conditions ](https://vueform.com/docs/validating-elements#and-conditions) and [OR Conditions](https://vueform.com/docs/validating-elements#or-conditions) but somehow it doesn't do what I need.

If I do this:
```html
<TextElement
    name="email"
    input-type="email"
    label="E-Mail-Adresse"
    :rules="[
      'email',
      {
        required: [['contact-type', ['E-Mail']]],
      },
    ]"
  />
```
then it would always require a valid email address and not allow `null`, even if the value is not required. So the AND doesn't seem to be what I'm looking for.

If I do this:
```html
:rules="[
    {
      required: [['contact-type', ['E-Mail']], ['email']],
    },
  ]"
```
then it would allow invalid email addresses.

Is there an example somewhere with conditional rules and valid phone numbers or valid email addresses ?

**Reply by adamberecz** [reply]
You can have the email rule conditional as well.

```js
:rules="[
  {
    required: [['contact-type', ['E-Mail']]],
  },
  {
    email: [['contact-type', ['E-Mail']]],
  }
]"
```

Actually it might be enough to set only the `email` because that doesn't allow `null` itself.

**Reply by julisch94** [reply]
omg, yes! Thank you so much. There was a misconception from my side. For some reason I thought the name `required` is an arbitrary name that I can select – when in fact it is the exact name of the rule that I want to apply conditionally.

Thanks for the quick response! ❤️
